buildpack.md

```diff
@@ -435,6 +446,60 @@ For each buildpack in the group, the lifecycle
 
 After analysis, the lifecycle MUST proceed to the build phase.
 
+
+## Phase #3: Image Extension
+
+### Purpose
+
+The purpose of the image extension phase is to apply dynamically-generated or static build-time and runtime modules that mutate the pre-build base image.
+
+During the image extension phase, an image extension might:
+
+1. Execute a pre-defined `Dockerfile` to install files in specific directories that require privilaged access.
+1. Dyanmically generate a `Dockerfile` based on the phase inputs and apply it.
+1. Execute a pre-defined `Dockerfile` to install system packages provided as input to the phase.
+
+### Process
+
+**GIVEN:**
+- The final ordered group of builimage extensions determined during the detection phase,
+- A directory containing application source code,
+- A `<output>` directory used to store generated artifacts
+- A shell, if needed,
+
+For each image extension in the group in order, the lifecycle MUST execute `/bin/build`.
+
+1. **If** the exit status of `/bin/build` is non-zero, \
+   **Then** the lifecycle MUST fail the phase.
+
+2. **If** the exit status of `/bin/build` is zero,
+   1. **If** there are additional image extensions in the group, \
+      **Then** the lifecycle MUST proceed to the next image extension's `/bin/build`.
+
+   2. **If** there are no additional image extension in the group, \
+      **Then** the lifecycle MUST proceed to the build phase.
+
+For each `/bin/build` executable in each image extension, the lifecycle:
+
+- MUST provide path arguments to `/bin/build` as described in the [Image Extension Interface](image-extension.md) section.
+- MUST configure the build environment as described in the [Environment](#environment) section.
+- MUST provide all `<plan>` entries that were required by any buildpack in the group during the detection phase with names matching the names that the buildpack provided.
+
+Correspondingly, each `/bin/build` executable:
+
+- MAY read from the `<app>` directory.
+- MUST NOT write to the `<app>` directory.
```

Comment:
```suggestion
- MUST NOT write to the `<app>` directory.
```

Is this one of those places where we hope they won't but can't do anything if they do?

Reply:
@jabrown85 i think it depends on the platform. Like, if it's running the extension in it's own container, it might be able to exclude `<app>` dir changes from it's snapshot. But if it's all running on the same filesystem/volume it won't be able to.

This isn't actually a suggested change right?

Reply:
Yeah, not a suggested change. I must have clicked the wrong button :)